src/languageSupport/languages/flux/lsp/connection.ts

```diff
@@ -126,112 +138,14 @@ class LspConnectionManager {
-  /// The change object only includes the replacement text, the position, and a little bit of editor-specific metadata.
-  /// The change object does NOT include anything which states whether or not the change is from an LSP-request,
-  /// (via the LSP applyEdit command).
-  /// As a result, we guess that any edit which is replacing the composition block is coming from an LSP request.
-  /// However, this heuristic fails for certain user-triggered events.
-  /// Such as a hotkey "undo" [cmd+z] of the last LSP applyEdit,
-  /// which generates a change object that looks exactly like an older applyEdit change.
-  _editorChangeIsFromLsp(change) {
-    return /^(from)(.|\n)*(\|> yield\(name: "_editor_composition"\)\n)$/.test(
-      change.text
-    )
-  }
-
-  _editorChangeIsWithinComposition(change) {
-    const compositionBlock = this._getCompositionBlockLines(
-      this._snapshot.read()
-    )
-    this._snapshot = this._model.createSnapshot()
-
-    if (!compositionBlock) {
-      return false
-    }
-    const {startLine, endLine} = compositionBlock
-
-    const changeInBlock =
-      change.range.startLineNumber >= startLine &&
-      change.range.endLineNumber <= endLine
-
-    const changeWithCompositionIdentifier =
-      change.text.includes(COMPOSITION_YIELD)
-
-    const isDeletion = change.text == ''
-    let deletionFromBlock = false
-    if (isDeletion) {
-      const linesDeleted =
-        change.range.endLineNumber - change.range.startLineNumber
-      deletionFromBlock =
-        change.range.startLineNumber >= startLine &&
-        change.range.endLineNumber <= endLine + linesDeleted
-    }
-
-    return changeInBlock || changeWithCompositionIdentifier || deletionFromBlock
-  }
-
-  _setEditorIrreversibleExit() {
-    this._model.onDidChangeContent(e => {
-      const shouldDiverge = e.changes.some(
-        change =>
-          this._editorChangeIsWithinComposition(change) &&
-          !this._editorChangeIsFromLsp(change)
-      )
-      if (shouldDiverge && !this._session.composition.diverged) {
-        event('Schema composition diverged - disable Flux Sync toggle')
-        this._callbackSetSession({
-          composition: {synced: false, diverged: true},
-        })
-      }
-    })
-
-    comments(this._editor, selection => {
-      const compositionBlock = this._getCompositionBlockLines(
-        this._model.getValue()
-      )
-      if (!compositionBlock) {
-        return
-      }
-      const {startLine, endLine} = compositionBlock
-      if (
-        selection.startLineNumber >= startLine &&
-        selection.endLineNumber <= endLine
-      ) {
-        this._callbackSetSession({
-          composition: {synced: false, diverged: true},
-        })
-      }
-    })
-  }
-
-  _compositionSyncStyle(startLine: number, endLine: number, synced: boolean) {
-    const classNamePrefix = synced
-      ? 'composition-sync--on'
-      : 'composition-sync--off'
```

Comment:
Seems like `composition-sync--off` and the related classes `composition-sync--off--first` and `composition-sync--off--last` can be removed in scss as well

Reply:
Good call. Removed.